transmogrifier/sources/json/aardvark.py

```diff
@@ -368,65 +369,75 @@ def get_publishers(source_record: dict) -> list[timdex.Publisher]:
                     for publisher in source_record["dct_publisher_sm"]
                 ]
             )
-        return publishers
+        return publishers or None
+
+    @classmethod
+    def get_provider(cls, source_record: dict) -> str | None:
+        return source_record.get("schema_provider_s") or None
 
-    @staticmethod
-    def get_rights(source: str, source_record: dict) -> list[timdex.Rights]:
-        """Get values from source record for TIMDEX rights field."""
-        rights = []
+    @classmethod
+    def get_rights(cls, source_record: dict, source: str) -> list[timdex.Rights] | None:
+        rights: list[timdex.Rights] = []
         kind_access_to_files = "Access to files"
+        rights.extend(cls._get_access_rights(source_record))
+        rights.extend(cls._get_license_rights(source_record))
+        rights.extend(cls._get_rights_and_rights_holders(source_record))
+        if source == "gisogm":
+            rights.extend(cls._get_gisogm_rights(kind_access_to_files))
+        elif source == "gismit":
+            rights.extend(cls._get_gismit_rights(source_record, kind_access_to_files))
+        return rights or None
```

Comment:
🔥 Looks great!  The sub-methods are clear, this higher level method is clear, easy to follow.